File: src/dlp/ast/sentences/Call.ts

import {Expression} from "../expressions/Expression";
import {AbstractLocatable} from "../AbstractLocatable";
import {Sentence} from "./Sentence";

export class Read extends AbstractLocatable implements Sentence,Expression {

    name: String;
    expressions: Expression[]

    constructor(name: String, expressions: Expression[], line: number, column: number){
        super(line, column);
        this.name = name;
        this.expressions = expressions;
    }
}

File: src/ast/sentences/Call.java
package ast.sentences;

import ast.AbstractLocatable;
import ast.expressions.Expression;
import java.util.List;

public class Call extends AbstractLocatable implements Sentence, Expression {

    private String name;
    private List<Expression> expressions;

    public Call(String name, List<Expression> expressions, int line, int column) {
        super(line, column);
        this.name = name;
        this.expressions = expressions;
    }

    public String getName() {
        return name;
    }

    public List<Expression> getExpressions() {
        return expressions;
    }
}
